src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -146,6 +161,11 @@ public ResponseEntity cancelAssessment(@PathVariable Long exerciseId, @PathVaria
         checkAuthorization(exercise);
 
         textBlockService.prepopulateFeedbackBlocks(result);
+
+        if (!authCheckService.isAtLeastInstructorForExercise(exercise)) {
+            result.getParticipation().filterSensitiveInformation(); // TODO: Check if this filter is correct
```

Comment:
Be careful when setting the student to `null` as we are in a transactional context here. The `@Transactional` annotation will persist the entity automatically when leaving this method, which might result in losing the connection of the participation to the student in the database.

Reply:
I removed the `@Transactional` here and adjusted the database call to avoid lazy initialization errors.